pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
-#: :class:`pywbem._logging.PywbemLoggers` methods that configure pywbem named
-#: loggers
-DEFAULT_LOG_DETAIL_LEVEL = 'min'
+* **Set the `detail_level` for pywbem log** - Because pywbem logs can
+  generate large quantities of information, pywbem allows the user to control
+  the quantity of information in each log.  Since this is a pywbem feature it
+  can only be configured by the pywbem :class:`~pywbem.WBEMConnection`
+  methods that configure loggers. Generally the detail levels are:
 
-#: Default maximum log entry size if none is supplied when creating
-#: :class:`~pywbem.LogOperationRecorder` objects.
-DEFAULT_MAX_LOG_ENTRY_SIZE = 1000
+  * `all` - All of the information is output.
 
-#: Default log destination if none is defined when named loggers are
-#: configured. 'none' means that there is no logging.
-DEFAULT_LOG_DESTINATION = 'none'
+  * `summary` - Only summary information is output. For api responses this
+    is primarily the count and type of objects returned. For http logs
+    the HTTP header information is output.
 
+  * `paths` - For api responses that include CIMInstanceNames or CIMClassNames,
+    (paths components in the objects) string representation of the
+    path components is output. For responses that  do not include CIMInstance
+    paths the complete objects representations are output.
 
-class MetaPywbemLoggers(type):
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
+  * :class:`py:int` - All of the information is output up to the size defined
+    by the integer value.  This applies to all log entries.
 
-    This metaclass allows the definition of __str__ and __repr__ at the
-    class level in the subclass.
-    """
-    loggers = {}
+* **Activate the loggers** - pywbem separates the configuration of the loggers
+  from activation to allow logger configuration by either pywbem methods or
+  standard python logging methods. Activation sets the pywbem loggers to a
+  state where they will generate log output when any of the WBEMConnection
+  methods that communicates with a WBEM server is called.  This involves
+  definining logging for a single connection. and optionally setting the
+  detail_level for the   log output
 
-    def __str__(cls):
-        return 'PywbemLoggers(loggers={s.loggers!r})'.format(s=cls)
+These loggers are only active for a :class:`~pywbem.WBEMConnection` object when:
 
-    def __repr__(cls):
-        return 'PywbemLoggers({s.loggers!r})'.format(s=cls)
+* A logger in the logger hierarchical name tree higher than or equal to the
+  pywbme loggers is defined before the connection is configured and the
+  logger is attached to :class:`~pywbem.WBEMConnection` object with one of the
+  :class:`~pywbem.WBEMConnection` methods.
+  Simply defining the loggers through standard python logger hander methods
+  does not activate these loggers since the logger is not attached to a
+  :class:`~pywbem.WBEMConnection` method
```

Comment:
Previously, we have introduced the concept of activating a logger. Now, we talk about attaching a logger. I suspect that is the same, and if so, we should stick with one terminology.

Reply:
Yes it is. activating a logger is the act of attaching it to a wbemconnection using the `connect` parameter of the various configure... methods.

Reply:
DONE